armada/handlers/armada.py

```diff
@@ -198,3 +202,26 @@ def show_diff(self, chart, installed_chart,
                 LOG.debug(line)
 
         return (len(chart_diff) > 0) or (len(values_diff) > 0)
+
+    def wait_until_deployed(self):
+        #
+        # TODO: configure command line arg to call function
+        #       command line arg for timeout 
+        #
+        os_pods = self.tiller.k8s.get_all_pods()
+        for pod in os_pods.items:
+            name = pod.metadata.name
+            status = pod.status.conditions[0].status
+            if status != "True":
+                LOG.info("{}-{}".format(name, status))
+                # wait for pod status to be True
+                #  check every second until timeout? 
+                default_timeout = 10
+                for x in range(0, default_timeout):
+                    sleep(1)
+                    status = pod.status.conditions[0].status
+                    if status == "True":
+                        break
+                    elif x == (default_timeout-1):
+                        sys.exit("Build timeout. Error causing chart: {} with message \"{}\""
```

Comment:
This should definitely be an exception and the logging, and potentially exiting, would be handled up the chain